### PR TITLE
[Lock] Expand `LockConflictedException` docs

### LINKED ISSUE #40969: [Lock] More explicit inline doc on `LockConflictedException`
**Description**
I made a silly error recently by ignoring `Lock::acquire()`'s return value, partly because – based on doc block hints – I was catching `LockConflictedException` and made the bad assumption from that exception's doc block description that it would be thrown and left uncaught in the case I was considering.

While the doc is accurate as far as the exception itself goes, I'm proposing a small addition to also highlight explicitly that it is caught by Symfony Lock for you, in the non-blocking case.

This particular route to confusion might be particular to me, but it seems like the benefits of having this hint highlighted wherever an IDE suggests to catch the exception could be of enough benefit to outweigh a bit more verbosity in the comments.

**Example**
N/A

### PR DESCRIPTION
**Pull request by NoelLH**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #40969
| License       | MIT
| Doc PR        | N/A

The goal of this tweak is to reduce the chance of bad assumptions about lock acquisition behaviour.

### CODE REVIEW

File: src/Symfony/Component/Lock/Exception/LockConflictedException.php [hunk 1]
```diff
@@ -11,9 +11,16 @@
...
 
+use Symfony\Component\Lock\Lock;
+
 /**
  * LockConflictedException is thrown when a lock is acquired by someone else.
  *
+ * In non-blocking mode it is caught by acquire() and acquireRead().
+ *
+ * @see Lock::acquire()
+ * @see Lock::acquireRead()
```
Comment (OskarStark) [inline src/Symfony/Component/Lock/Exception/LockConflictedException.php]:
```suggestion
 * In non-blocking mode it is caught by {@see Lock::acquire()} and {@see Lock::acquireRead()}.
```
?

Comment (NoelLH) [reply src/Symfony/Component/Lock/Exception/LockConflictedException.php]: Ah yes, thanks!

**Comment by derrabus** [comment]
PHPDoc changes are no new features. This change needs to target the 4.4 branch instead.

**Comment by NoelLH** [comment]
> PHPDoc changes are no new features. This change needs to target the 4.4 branch instead.

Ah, got it. I'll try to work out the right way to do that another day.

**Comment by NoelLH** [comment]
Rebasing looks like it'll take an eternity so I'll make a new branch.

**Pull request closed without merging**